### PR TITLE
fix: sanitize git branch name for use in docker image tags

### PR DESCRIPTION
**Pull request by notnmeyer**
works on this branch,
image

only handles forward slashes for now, but easy to modify later.

this also replaces using  `Ilshidur/action-slack` with a task, `docker:slack-notifier`. doing the slack notification ourselves means that we can sanitize the tag within Task and not worry about exposing it back to the gha workflow for consumption in other steps.

### CODE REVIEW

File: .github/workflows/deployDemo.yml [hunk 1]
```diff
@@ -47,7 +47,9 @@ jobs:
       - run: |
           echo "DOCKER_REGISTRY=${{ steps.login-ecr.outputs.registry }}" >> $GITHUB_ENV
           echo "REPO=demo" >> $GITHUB_ENV
-          echo "TAG=${BRANCH}-b${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
+          TAG=$(TAG=${BRANCH}-b${GITHUB_RUN_NUMBER} task docker:clean-tag)
```
Comment (notnmeyer) [inline .github/workflows/deployDemo.yml:50]: i really dont love this. the only reason to have set the "corrected" tag back to the gha env is to reference the artifact in the slack notification that follows. if we did the slack notification ourselves in Task we could drop this entirely.

**Comment by notnmeyer** [comment]
marinating on this for a bit in case i come up with something better

File: bin/clean-docker-tag [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+set -eu
+# replaces character that aren't valid in Docker image tags with hyphens
+echo $1 | sed 's/\//-/g'
```
Comment (notnmeyer) [inline bin/clean-docker-tag:4]: 🍖

**Review by devdoshi — APPROVED** [review]

File: workspaces/ui/Taskfile.yml [hunk 3]
```diff
@@ -16,18 +16,26 @@ tasks:
...
-    - docker push $DOCKER_REGISTRY/$REPO:$TAG
-    - |
-      if [[ "$TAG" =~ ^release-b ]]
-      then
-        docker tag $DOCKER_REGISTRY/$REPO:$TAG $DOCKER_REGISTRY/$REPO:latest
-        docker push $DOCKER_REGISTRY/$REPO:latest
-      fi
+    vars:
+      TAG:
+        sh: echo $(../../bin/clean-docker-tag $TAG)
```
Comment (notnmeyer) [inline workspaces/ui/Taskfile.yml:29]: the var is set at the task level to not pollute the global scope. invoking any task through the top-level taskfile will load all included taskfiles and add global scope vars/env.

**Pull request merged**